BackendBench/backends/llm.py

```diff
@@ -167,14 +167,14 @@ def test_kernel_correctness(
             kernel_file = os.path.join(op_dir, f"{op_name}_implementation_v{attempt}.py")
 
             if not os.path.exists(kernel_file):
-                is_triton = "triton.jit" in kernel_code or "@triton.jit" in kernel_code
-                if is_triton:
-                    full_code = self._prepare_triton_code(kernel_code)
-                else:
-                    full_code = self._prepare_torch_code(kernel_code)
-
-                with open(kernel_file, "w") as f:
-                    f.write(full_code)
+                module_name = f"{op_name}_implementation_v{attempt}"
+                compile_kernel_from_string(
```

Comment:
Thanks for the catch, and the quick fix :)

This was a bit confusing for me to read, so can you do a quick refactor.

Split https://github.com/meta-pytorch/BackendBench/blob/64bfe12836024c464e40239be20e5c26ca7fd244/BackendBench/utils.py#L284 into two functions one which does the file writing (the side effect we want here), and one that grabs the kernel function. 

In practice we should just refactor this function to use `compile_kernel_from_string`, but we can do that later.